iothub/device/tests/Pipeline/RetryDelegatingHandlerTests.cs

```diff
@@ -454,6 +416,190 @@ public async Task RetrySetRetryPolicyVerifyInternalsSuccess()
             retryPolicy.Counter.Should().Be(2);
         }
 
+        [Ignore]
+        [TestMethod]
+        public async Task RetryDelegatingHandler_CloseAsync_CancelsPendingOpenAsync()
+        {
+            // arrange
+            var contextMock = new PipelineContext
+            {
+                RetryPolicy = new TestRetryOncePolicy(),
+                ConnectionStatusChangeHandler = (connectionStatusInfo) => { }
+            };
+
+            var nextHandlerMock = new Mock<IDelegatingHandler>();
+            nextHandlerMock
+                .Setup(x => x.OpenAsync(It.IsAny<CancellationToken>()))
+                .Returns(async (CancellationToken ct) =>
+                {
+                    // Artificial wait to simulate the transport protocol level operations.
+                    await Task.Delay(TimeSpan.FromSeconds(5), ct);
+                });
+
+            using var retryDelegatingHandler = new RetryDelegatingHandler(contextMock, nextHandlerMock.Object);
+
+            // act
+            Task openTask = retryDelegatingHandler.OpenAsync(CancellationToken.None);
+            await retryDelegatingHandler.CloseAsync(CancellationToken.None).ConfigureAwait(false);
+
+            // assert
+
+            Func<Task> openFunc = async () => await openTask.ConfigureAwait(false);
+            await openFunc.Should().ThrowAsync<OperationCanceledException>();
+            nextHandlerMock.Verify(
+                x => x.OpenAsync(It.IsAny<CancellationToken>()),
+                Times.Once);
+        }
+
+        [Ignore]
```

Comment:
Marking this test as ignored for now since this functionality hasn't been added yet